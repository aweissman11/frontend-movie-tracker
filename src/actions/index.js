export const getUserLoggedIn = (id, name) => {
  return {
    type: 'GET_USER_LOGGED_IN',
    user: { id, name }
  }
}

export const logUserOut = () => {
  return {
    type:  'LOG_USER_OUT'
  }
}

export const displayLogin = () => {
  return {
    type: 'DISPLAY_LOGIN'
  }
}

export const displaySignUp = () => {
  return {
    type: 'DISPLAY_SIGN_UP'
  }
}

export const setFavorites = (favorites = []) => {
  return {
    type: 'SET_FAVORITES',
    favorites
  }
}

export const updateFilters = (filters) => {
  return {
    type: 'UPDATE_FILTERS',
    filters
  }
}
export const updateSearchQuery = (searchQuery) => {
  return {
    type: 'UPDATE_QUERY',
    searchQuery
  }
}

export const setMovieList = (movies) => {
  return {
    type: 'SET_MOVIE_LIST',
    movies
  }
}

export const setHasErrored = (bool) => {
  return {
    type: 'SET_HAS_ERRORED',
    hasErrored: bool
  }
}

export const setIsOk = (bool) => {
  return {
    type: 'SET_IS_NOT_OK',
    notOk: bool
  }
}

export const isLoading = (bool) => {
  return {
    type: 'IS_LOADING',
    isLoading: bool
  }
}