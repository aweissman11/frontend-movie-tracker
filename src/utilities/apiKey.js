export const apiKey = 'db393102c6d38bb71fed129f3aaf36d8';