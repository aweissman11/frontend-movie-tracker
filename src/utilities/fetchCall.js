const fetchCall = async (url) => {
  const response = await fetch(url);
  if (!response.ok) {
    console.error(Error(response.statusText));
    return 'failed'
  }
  return await response.json();
}

export default fetchCall;