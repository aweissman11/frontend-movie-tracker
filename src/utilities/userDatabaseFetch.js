export const checkUserList = async (data) => {
  const url = 'http://localhost:3000/api/users'

  const optionsObject = {
    method: "POST", 
    body: JSON.stringify(data),
    credentials: "same-origin", 
    headers: {
      "Content-Type": "application/json",
    }
  }

  const response = await fetch(url, optionsObject);
  return await response.json();
}

export const createNewUser = async (data) => {
  const url = 'http://localhost:3000/api/users/new'

  const optionsObject = {
    method: "POST", 
    body: JSON.stringify(data),
    credentials: "same-origin", 
    headers: {
        "Content-Type": "application/json",
    }
  }

  const response = await fetch(url, optionsObject);
  return await response.json();
}

export const addFavorite = async (data) => {
  const url = 'http://localhost:3000/api/users/favorites/new'

  const optionsObject = {
    method: "POST", 
    body: JSON.stringify(data),
    credentials: "same-origin", 
    headers: {
        "Content-Type": "application/json",
    }
  }

  const response = await fetch(url, optionsObject);
  return await response.json();
}

export const removeFavorite = async (userId, movieId) => {
  const url = `http://localhost:3000/api/users/${userId}/favorites/${movieId}`

  const optionsObject = {
    method: "DELETE", 
    body: JSON.stringify({
      user_id: userId,
      movie_id: movieId
    }),
    credentials: "same-origin", 
    headers: {
        "Content-Type": "application/json",
    }
  }

  const response = await fetch(url, optionsObject);
  return await response.json();
}

