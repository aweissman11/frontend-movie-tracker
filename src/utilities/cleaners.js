import { apiKey } from "./apiKey";

export const getMoviePosterUrl = (movies) => {
  return movies.map( movie => {
    return movie.poster_path
  })
}

export const getTodaysDate = () => {
  const today = new Date();
  let dd = today.getDate();
  let mm = today.getMonth()+1;
  let yyyy = today.getFullYear();
  if(dd<10) {
      dd = '0'+dd;
  } 
  if(mm<10) {
      mm = '0'+mm;
  } 
  return `${mm}-${dd}-${yyyy}`;
}

export const getFilterUrl = (filterProperties) => {
  const today = getTodaysDate();

  let url = `https://api.themoviedb.org/3/discover/movie?api_key=${apiKey}&language=en-US?primary_release_date.lte=${today}&sort_by=popularity.desc`

  const { genre, year, rating, sort } = filterProperties;
  if (genre && genre !== 'GENRE') {
    url += `&with_genres=${genre}`
  }
  
  if (year && year !== 'YEAR') {
    url += `&year=${year}`
  }

  if (rating && rating !== 'RATING') {
    url += `&certification_country=US&certification=${rating}`
  }

  if (sort && sort !== 'SORT') {
    url += `&sort_by=${sort}`
  }

  return url;

}