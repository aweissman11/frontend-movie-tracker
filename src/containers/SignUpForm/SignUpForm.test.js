/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import { connect } from 'react-redux';
import * as userDatabaseFetch from '../../utilities/userDatabaseFetch';

import { SignUpForm, mapDispatchToProps, mapStateToProps } from './index';
import * as Actions from '../../actions';

describe('SignUpForm', () => {
  let wrapper;
  const mockUser = {
    id: 1
  }
  let mockShowSignUp = true;
  let mockLogUserIn;
  let mockSetIsLoading;
  let mockDisplayLogin;
  let mockSubscribe;
  let mockDispatch;
  let mockGetState;
  let mockUserDatabaseFetch = {
    checkUserList: jest.fn().mockImplementation(() => {
      return Promise.resolve({
        data: {
          id: 1,
          name: 'mock name'
        }
      });
    })
  }


  const mockStore = {
    subscribe: mockSubscribe,
    dispatch: mockDispatch,
    getState: mockGetState
  };

  const defaultState = {
    name: '',
    email: '',
    password: '',
    confirmPassword: '',
    signUpError: '',
    userDatabaseFetch: userDatabaseFetch,
    activeErrorText: 'email address already registered'
  };

  const mockPreventDefault = jest.fn();

  const mockEvent = {
    target: {
      name: 'email',
      value: 'email@email.com'
    },
    preventDefault: mockPreventDefault
  }

  beforeEach(() => {
    mockLogUserIn = jest.fn();
    mockSetIsLoading = jest.fn();
    mockDisplayLogin = jest.fn();
    mockSubscribe = jest.fn();
    mockDispatch = jest.fn();
    mockGetState = jest.fn();

    wrapper = shallow(<SignUpForm 
      store={mockStore}
      user={mockUser}
      showSignUp={mockShowSignUp}
      logUserIn={mockLogUserIn}
      displayLogin={mockDisplayLogin}
      setIsLoading={mockSetIsLoading}
    />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state', () => {
    expect(wrapper.state()).toEqual(defaultState)
  });

  it('should call displayLogin on click', () => {
    wrapper.find('.skip-sign-up-button').simulate('click', mockEvent);

    expect(mockDisplayLogin).toHaveBeenCalled();

  });

  describe('handleChange', () => {
    it('should call set state', async () => {
      const expected = 'email@email.com'
      await wrapper.instance().handleChange(mockEvent);

      expect(wrapper.state().email).toEqual(expected);
    });

    it('should call handleChange on email input', () => {
      const spy = spyOn(wrapper.instance(), 'handleChange')
      wrapper.instance().forceUpdate();

      wrapper.find('.email-input').simulate('change', mockEvent);

      expect(spy).toHaveBeenCalledWith(mockEvent);
    });

    it('should call handleChange on password input', () => {
      const spy = spyOn(wrapper.instance(), 'handleChange')
      wrapper.instance().forceUpdate();

      wrapper.find('.password-input').simulate('change', mockEvent);

      expect(spy).toHaveBeenCalledWith(mockEvent);
    });

    it('should call handleChange on confirm password input', () => {
      const spy = spyOn(wrapper.instance(), 'handleChange')
      wrapper.instance().forceUpdate()

      wrapper.find('.password-confirm-input').simulate('change', mockEvent);

      expect(spy).toHaveBeenCalledWith(mockEvent);
    });

    it('should call handleChange on name input', () => {
      const spy = spyOn(wrapper.instance(), 'handleChange');
      wrapper.instance().forceUpdate()

      wrapper.find('.name-input').simulate('change', mockEvent);

      expect(spy).toHaveBeenCalledWith(mockEvent);
    });
  });

  describe('removeWarning', () => {
    it('should call set state', async () => {
      const expected = '';

      await wrapper.instance().removeWarning();

      expect(wrapper.state().signUpError).toEqual(expected);
    });
  })

  describe('userWarning', () => {
    it('should call set state', async () => {
      const mockErrorText = 'activeErrorText'
      const mockWarning = 'mock warning'
      const mockSignUpError = 'sign-up-error-active'

      await wrapper.instance().userWarning(mockWarning);

     expect(wrapper.state()[mockErrorText]).toEqual(mockWarning);
     expect(wrapper.state().signUpError).toEqual(mockSignUpError);
    });

  });

  describe('createNewUser', () => {

    const mockCreateNewUser = jest.fn().mockImplementation(() => {
      return Promise.resolve({
        data: {
          id: 1,
          name: 'mock name'
        }
      });
    });

    const mockUserDataBaseFetch = {
      createNewUser: mockCreateNewUser
    }

    const mockEmail = 'mock email';
    const mockPassword = 'mock password';
    const mockName = 'mock name'

    beforeEach(async () => {
      await wrapper.setState({
        email: mockEmail,
        password: mockPassword,
        name: mockName,
        userDatabaseFetch: mockUserDataBaseFetch
      });
    });

    it('should call preventDefault', async () => {
      await wrapper.instance().createNewUser(mockEvent);

      expect(mockPreventDefault).toHaveBeenCalled();
    });

    it('should call userWarning with the correct params when an error response is returned', async () => {

      const spy = spyOn(wrapper.instance(), 'userWarning');
      wrapper.instance().forceUpdate();
      const mockErrorResponse = jest.fn().mockImplementation(() => {
        return Promise.resolve({error: 'error'})
      });

      await wrapper.instance().createNewUser(mockEvent);

      expect(spy).toHaveBeenCalled();
    });

    it('should call createNewUser on submit', () => {
      const spy = spyOn(wrapper.instance(), 'createNewUser')
      wrapper.instance().forceUpdate();

      wrapper.find('form').simulate('submit');

      expect(spy).toHaveBeenCalled();
    })

  });

});