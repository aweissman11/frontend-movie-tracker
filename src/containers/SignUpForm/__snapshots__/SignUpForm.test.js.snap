// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUpForm
    displayLogin={[MockFunction]}
    logUserIn={[MockFunction]}
    showSignUp={true}
    store={
      Object {
        "dispatch": [MockFunction],
        "getState": [MockFunction],
        "subscribe": [MockFunction],
      }
    }
    user={
      Object {
        "id": 1,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          <span>
            movie
          </span>
          Tracker
        </h1>,
        <h4>
          for the love of film
        </h4>,
        <div
          className="login-form-logo-wrapper"
        >
          <Logo />
        </div>,
        <input
          className="name-input"
          name="name"
          onChange={[Function]}
          placeholder="name"
          value=""
        />,
        <img
          alt="user"
          className="user-icon"
          src="./user.png"
        />,
        <input
          className="email-input"
          name="email"
          onChange={[Function]}
          pattern="^\\\\w+@[a-zA-Z_]+?\\\\.[a-zA-Z]{2,3}$"
          placeholder="email"
          title="Enter a Valid Email Address"
          value=""
        />,
        <img
          alt="mail"
          className="email-icon"
          src="./email.png"
        />,
        <input
          className="password-input"
          name="password"
          onChange={[Function]}
          placeholder="password"
          type="password"
          value=""
        />,
        <img
          alt="lock"
          className="password-icon"
          src="./password.png"
        />,
        <img
          alt="lock"
          className="password-confirm-icon"
          src="./password.png"
        />,
        <input
          className="password-confirm-input"
          name="confirmPassword"
          onChange={[Function]}
          placeholder="confirm password"
          type="password"
          value=""
        />,
        <input
          className="create-user-btn"
          type="submit"
          value="sign up"
        />,
        <Link
          className="skip-sign-up"
          replace={false}
          to="/login"
        >
          <button
            className="skip-sign-up-button"
            onClick={[MockFunction]}
          >
            already a user
          </button>
        </Link>,
        <div
          className="signup-error-wrapper "
        >
          <p>
            email address already registered
          </p>
        </div>,
      ],
      "className": "sign-up-form undefined",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              movie
            </span>,
            "Tracker",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "movie",
            },
            "ref": null,
            "rendered": "movie",
            "type": "span",
          },
          "Tracker",
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "for the love of film",
        },
        "ref": null,
        "rendered": "for the love of film",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Logo />,
          "className": "login-form-logo-wrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "name-input",
          "name": "name",
          "onChange": [Function],
          "placeholder": "name",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "user",
          "className": "user-icon",
          "src": "./user.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "email-input",
          "name": "email",
          "onChange": [Function],
          "pattern": "^\\\\w+@[a-zA-Z_]+?\\\\.[a-zA-Z]{2,3}$",
          "placeholder": "email",
          "title": "Enter a Valid Email Address",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "mail",
          "className": "email-icon",
          "src": "./email.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "password-input",
          "name": "password",
          "onChange": [Function],
          "placeholder": "password",
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "lock",
          "className": "password-icon",
          "src": "./password.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "lock",
          "className": "password-confirm-icon",
          "src": "./password.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "password-confirm-input",
          "name": "confirmPassword",
          "onChange": [Function],
          "placeholder": "confirm password",
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "create-user-btn",
          "type": "submit",
          "value": "sign up",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button
            className="skip-sign-up-button"
            onClick={[MockFunction]}
          >
            already a user
          </button>,
          "className": "skip-sign-up",
          "replace": false,
          "to": "/login",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "already a user",
            "className": "skip-sign-up-button",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "already a user",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p>
            email address already registered
          </p>,
          "className": "signup-error-wrapper ",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "email address already registered",
          },
          "ref": null,
          "rendered": "email address already registered",
          "type": "p",
        },
        "type": "div",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            <span>
              movie
            </span>
            Tracker
          </h1>,
          <h4>
            for the love of film
          </h4>,
          <div
            className="login-form-logo-wrapper"
          >
            <Logo />
          </div>,
          <input
            className="name-input"
            name="name"
            onChange={[Function]}
            placeholder="name"
            value=""
          />,
          <img
            alt="user"
            className="user-icon"
            src="./user.png"
          />,
          <input
            className="email-input"
            name="email"
            onChange={[Function]}
            pattern="^\\\\w+@[a-zA-Z_]+?\\\\.[a-zA-Z]{2,3}$"
            placeholder="email"
            title="Enter a Valid Email Address"
            value=""
          />,
          <img
            alt="mail"
            className="email-icon"
            src="./email.png"
          />,
          <input
            className="password-input"
            name="password"
            onChange={[Function]}
            placeholder="password"
            type="password"
            value=""
          />,
          <img
            alt="lock"
            className="password-icon"
            src="./password.png"
          />,
          <img
            alt="lock"
            className="password-confirm-icon"
            src="./password.png"
          />,
          <input
            className="password-confirm-input"
            name="confirmPassword"
            onChange={[Function]}
            placeholder="confirm password"
            type="password"
            value=""
          />,
          <input
            className="create-user-btn"
            type="submit"
            value="sign up"
          />,
          <Link
            className="skip-sign-up"
            replace={false}
            to="/login"
          >
            <button
              className="skip-sign-up-button"
              onClick={[MockFunction]}
            >
              already a user
            </button>
          </Link>,
          <div
            className="signup-error-wrapper "
          >
            <p>
              email address already registered
            </p>
          </div>,
        ],
        "className": "sign-up-form undefined",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                movie
              </span>,
              "Tracker",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "movie",
              },
              "ref": null,
              "rendered": "movie",
              "type": "span",
            },
            "Tracker",
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "for the love of film",
          },
          "ref": null,
          "rendered": "for the love of film",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Logo />,
            "className": "login-form-logo-wrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "name-input",
            "name": "name",
            "onChange": [Function],
            "placeholder": "name",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "user",
            "className": "user-icon",
            "src": "./user.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "email-input",
            "name": "email",
            "onChange": [Function],
            "pattern": "^\\\\w+@[a-zA-Z_]+?\\\\.[a-zA-Z]{2,3}$",
            "placeholder": "email",
            "title": "Enter a Valid Email Address",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "mail",
            "className": "email-icon",
            "src": "./email.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "password-input",
            "name": "password",
            "onChange": [Function],
            "placeholder": "password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "lock",
            "className": "password-icon",
            "src": "./password.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "lock",
            "className": "password-confirm-icon",
            "src": "./password.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "password-confirm-input",
            "name": "confirmPassword",
            "onChange": [Function],
            "placeholder": "confirm password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "create-user-btn",
            "type": "submit",
            "value": "sign up",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button
              className="skip-sign-up-button"
              onClick={[MockFunction]}
            >
              already a user
            </button>,
            "className": "skip-sign-up",
            "replace": false,
            "to": "/login",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "already a user",
              "className": "skip-sign-up-button",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": "already a user",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
              email address already registered
            </p>,
            "className": "signup-error-wrapper ",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "email address already registered",
            },
            "ref": null,
            "rendered": "email address already registered",
            "type": "p",
          },
          "type": "div",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
